pkg/agent/util/winfirewall/winfirewall.go

```diff
@@ -0,0 +1,174 @@
+// +build windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package winfirewall
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+)
+
+type FWRuleDirection string
+
+const (
+	FWRuleIn  FWRuleDirection = "Inbound"
+	FWRuleOut FWRuleDirection = "Outbound"
+)
+
+type fwRuleAction string
+
+const (
+	fwRuleAllow fwRuleAction = "Allow"
+	fwRuleDeny  fwRuleAction = "Block"
+)
+
+type fwRuleProtocol string
+
+const (
+	fwRuleIPProtocol  fwRuleProtocol = "Any"
+	fwRuleTCPProtocol fwRuleProtocol = "TCP"
+	fwRuleUDPProtocol fwRuleProtocol = "UDP"
+)
+
+const (
+	fwRuleGroup string = "Antrea"
+)
+
+type winFirewallRule struct {
+	name          string
+	action        fwRuleAction
+	direction     FWRuleDirection
+	protocol      fwRuleProtocol
+	localAddress  *net.IPNet
+	remoteAddress *net.IPNet
+	localPorts    []uint16
+	remotePorts   []uint16
+}
+
+// add adds Firewall rule on the Windows host. The name and display name of the firewall rule are the same.
+func (r *winFirewallRule) add() error {
+	cmd := fmt.Sprintf("New-NetFirewallRule -Enabled True -DisplayGroup %s %s", fwRuleGroup, r.getCommandString())
```

Comment:
Could you double check "-DisplayGroup" is a valid param? I can find this param in doc:
https://docs.microsoft.com/en-us/powershell/module/netsecurity/new-netfirewallrule?view=win10-ps